Lib/shutil.py

```diff
@@ -1441,3 +1441,118 @@ def which(cmd, mode=os.F_OK | os.X_OK, path=None):
                 if _access_check(name, mode):
                     return name
     return None
+
+def _create_or_replace(dst, create_temp_dst):
+    """Create or overwrite file `dst` atomically via os.replace.
+
+    The file to replace `dst` is first created with a temporary pathname.
+
+    `create_temp_dst` is a callable taking a single argument: a pathname
+    where the temporary file to replace `dst` should be created.  If it
+    raises FileExistsError, it will be called again with another temp pathname.
+    """
+    # Following import can't be at module level: to build standard modules,
+    # setup.py uses argparse which uses shutil.get_terminal_size().
+    # tempfile imports random which imports math which isn't available
+    import tempfile
+    try:
+        # Loop until successful creation of previously non-existent temp pathname
+        while True:
```

Comment:
Can this use `mkdtemp` and create the temp file inside of that dir, instead of the loop?